mongosql/src/algebrizer/definitions.rs

```diff
@@ -475,14 +478,105 @@ impl<'a> Algebrizer<'a> {
         }
 
         // Build the Project Stage using the source and built expression.
-        let stage = mir::Stage::Project(mir::Project {
+        let project_stage = mir::Stage::Project(mir::Project {
             source: Box::new(source),
-            expression,
+            expression: expression.clone(),
             is_add_fields,
             cache: SchemaCache::new(),
         });
-        stage.schema(&self.schema_inference_state())?;
-        Ok(stage)
+        project_stage.schema(&self.schema_inference_state())?;
+        Ok(if is_distinct {
+            let group_stage = Self::build_distinct_values_group_and_project_stage(
```

Comment:
[nit] this will be a `Project` stage with a `Group` source so the name `group_stage` is outdated. Maybe we could call this variable `distinct_stages` or something like that?

Reply:
That works, I renamed it to `distinct_stages`.